### DISCUSSION TITLE
SEEED XIAO ESP32C3 : IDF 5.04 build does not start with preview version

### DISCUSSION THREAD

**Discussion by ricksorensen** [micropython/micropython#15700] [ESP32]
I am not able to get the  ESP32_GENERIC_C3 to run on a SEEED XIAO ESP32C3 with the current preview version:

TLDR: 2024-08-22 preview download does not run, but if compiled with IDF 5.2.2 it does run. 
First failed (I think) revision: 0a11832 ... `esp32: Use the ESP-IDF default esp_console config for ESP32-C3` which has many usb related modifications.

[v1.24.0-preview.224.g6c3dc0c0b (2024-08-22) .bin](https://micropython.org/resources/firmware/ESP32_GENERIC_C3-20240822-v1.24.0-preview.224.g6c3dc0c0b.bin) loads but hangs mpremote with no response over USB.

The 1.23 release version runs fine:
[v1.23.0 (2024-06-02) .bin](https://micropython.org/resources/firmware/ESP32_GENERIC_C3-20240602-v1.23.0.bin)

When I compile 1.23 on my system with IDF 5.04 it loads and runs fine.  When I compile the current (2024-08-22) version with IDF 5.04 it compiles and loads, but does not run (see above).
Whe I compile the current (2024-08-22) version with IDF 5.2.2, it compiles, loads, and runs.

[v1.24.0-preview.213.g76dd4facb (2024-08-20) .bin](https://micropython.org/resources/firmware/ESP32_GENERIC_C3-20240820-v1.24.0-preview.213.g76dd4facb.bin) hangs mpremote also.

The versions early in August worked okay, but I think the run broke in the last week or so.  I see there were several changes some impacting the USB connection.  First failed (I think) revision: https://github.com/micropython/micropython/commit/0a11832cddbeed57bf0267afa0c833584817312d ... esp32: Use the ESP-IDF default esp_console config for ESP32-C3 which has many usb related modifications.

Maybe I have an IDF 5.04  configuration issue?

**Comment by jkorte-dev** [comment]
I tried the preview build 2 or 3 days ago on a lolin c3 mini and could not get a usb repl with mpremote . 1.23 release worked fine

**Reply by ricksorensen** [reply]
@jkorte-dev see issue #15701
